src/graphql/types/scalars/ln-invoice-payment-request.ts

```diff
@@ -0,0 +1,24 @@
+import { GT } from "@graphql/index"
+
+const LnInvoicePaymentRequest = new GT.Scalar({
+  name: "LnInvoicePaymentRequest",
+  parseValue(value) {
+    return validLnInvoicePaymentRequest(value)
+  },
+  parseLiteral(ast) {
+    if (ast.kind === GT.Kind.STRING) {
+      return validLnInvoicePaymentRequest(ast.value)
+    }
+    return new Error("Invalid type for LnInvoicePaymentRequest")
+  },
+})
+
+function validLnInvoicePaymentRequest(value) {
+  // TODO: verify/improve
+  if (value.match(/^ln[A-Za-z0-9]+$/)) {
```

Comment:
Could we validate by calling the decode domain function?

Reply:
I guess we could but might be an overkill (during GraphQL parsing). Other input errors might block the operation anyway.

Not sure... @nicolasburtey what do you think?